File: 2014/nazionali/maree/att/grader.pas
uses maree;

const MAXM = 3000000;
var i, N, M, T : longint;
    S, E       : array[0..MAXM-1] of longint;
    fin, fout  : text;

begin
    assign(fin, 'input.txt'); reset(fin);
    assign(fout, 'output.txt'); rewrite(fout);
    read(fin, N, M, T);
    for i:=0 to M-1 do read(fin, S[i], E[i]);
    writeln(fout, solve(N, M, T, S, E));
    close(fin);
    close(fout);
end.

File: 2014/nazionali/maree/att/grader.c
#include <stdio.h>
#include <assert.h>
#define MAXM 3000000

static int i, N, M, T;
static int S[MAXM], E[MAXM];

int solve(int N, int M, int T, int* S, int* E);

static FILE *fin, *fout;

int main(){
    fin = fopen("input.txt", "r");
    fout = fopen("output.txt", "w");
    assert(3 == fscanf(fin, "%d%d%d", &N, &M, &T));
    for(i=0; i<M; i++) assert(2 == fscanf(fin, "%d%d", S+i, E+i));
    fprintf(fout, "%d\n", solve(N, M, T, S, E));
    fclose(fin);
    fclose(fout);
    return 0;
}
